sklearn/preprocessing/_discretization.py

```diff
@@ -236,20 +241,26 @@ def fit(self, X, y=None):
 
         return self
 
-    def _validate_n_bins(self, n_features):
+    def _validate_n_bins(self, n_features, n_samples):
         """Returns n_bins_, the number of bins per feature.
         """
         orig_bins = self.n_bins
-        if isinstance(orig_bins, numbers.Number):
+        if isinstance(orig_bins, numbers.Number) | isinstance(orig_bins, str):
+            if orig_bins == 'auto':
+                # calculcate number of bins
+                # depending on number of samples with Sturges rule
+                orig_bins = int(np.ceil(np.log2(n_samples) + 1.))
```

Comment:
I wonder whether +1 is actually beneficial for discretisation. The number of bins looks fairly big for smallish numbers.

Reply:
It is needed for very small number of samples:
log2(1) = 0 but log2(1) + 1 = 1
log2(2) = 1 but log2(2) + 1 = 2
etc.
For larger n_samples adding 1 will not drastically change the output. So I would vote to keep it